File: app/src/main/java/com/nuhkoca/offlineapp/di/component/BindingComponent.java
package com.nuhkoca.offlineapp.di.component;


import com.nuhkoca.offlineapp.di.module.BindingModule;
import com.nuhkoca.offlineapp.di.qualifier.DataBinding;

import androidx.databinding.DataBindingComponent;
import dagger.Component;

@DataBinding
@Component(dependencies = AppComponent.class, modules = BindingModule.class)
public interface BindingComponent extends DataBindingComponent {
}

File: app/src/main/java/com/nuhkoca/offlineapp/di/component/BindingComponent.kt
package com.nuhkoca.offlineapp.di.component

import androidx.databinding.DataBindingComponent
import com.nuhkoca.offlineapp.binding.DateParsingBindingAdapter
import com.nuhkoca.offlineapp.binding.ImageBindingAdapter
import com.nuhkoca.offlineapp.binding.ListBindingAdapter
import com.nuhkoca.offlineapp.di.module.BindingModule
import com.nuhkoca.offlineapp.di.scope.DataBinding
import dagger.Component

@DataBinding
@Component(dependencies = [AppComponent::class], modules = [BindingModule::class])
interface BindingComponent: DataBindingComponent {

    override fun getImageBindingAdapter(): ImageBindingAdapter?

    override fun getListBindingAdapter(): ListBindingAdapter?

    override fun getDateParsingBindingAdapter(): DateParsingBindingAdapter?
}